File: 프로그래머스/unrated/148653. 마법의 엘리베이터/마법의 엘리베이터.py
def solution(storey):
    cnt = 0
    while storey:
        num = storey%10
        storey = storey//10
        if num > 5 or (num == 5 and storey%10 >= 5):
            cnt += 10 - num
            storey+=1
        else:
            cnt+=num
    return cnt

File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.js
function solution(storey) {
  let ans = 0;
  let num = storey.toString().split("");

  for (let i = num.length - 1; i >= 0; i--) {
    let current = Number(num[i]);
    if (current < 5) {
      ans += current;
    } else if (current > 5) {
      ans += 10 - current;
      if (i > 0) num[i - 1] = Number(num[i - 1]) + 1;
      else ans++;
    } else {
      if (i === 0) {
        ans += 5;
      } else {
        if (num[i - 1] >= 5) {
          ans += 5;
          num[i - 1]++;
        } else {
          ans += 5;
        }
      }
    }
  }

  return ans;
}
